### PR TITLE
Rewrite [travisphpversion]; improve error message in expectBadge

### PR DESCRIPTION
**Pull request by paulmelnikow**
The change in `expectBadge` prints a more helpful error when `message` is empty.

Ref #2863

### CODE REVIEW

File: core/service-test-runner/icedfrisby-shields.js [hunk 1]
```diff
@@ -46,11 +46,7 @@ const factory = superclass =>
       if (typeof expected === 'string') {
         expect(json[name], `${name} mismatch`).to.equal(expected)
       } else if (typeof expected === 'object') {
-        Joi.validate(json[name], expected, err => {
-          if (err) {
-            throw err
-          }
-        })
+        Joi.attempt(json[name], expected, `${name} mismatch:`)
```
Comment (calebcartwright) [inline core/service-test-runner/icedfrisby-shields.js]: 👍

File: services/travis/travis-php-version.service.js [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
+      php: Joi.array().items(optionalNumberOrString),
+      matrix: Joi.object({
+        include: Joi.array().items(Joi.object({ php: optionalNumberOrString })),
+      }),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class TravisPhpVersion extends BaseJsonService {
+  constructor(context, config) {
```
Comment (calebcartwright) [inline services/travis/travis-php-version.service.js]: May want to go ahead and move this down between `render` and `transform` given #3353

**Comment by calebcartwright** [comment]
Do you have a `tester` for the redirector pending? If not I can add one

**Comment by paulmelnikow** [comment]
Ah, nope, that'd be great!

**Pull request merged**